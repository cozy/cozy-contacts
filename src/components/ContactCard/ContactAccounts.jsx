import React from 'react'
import PropTypes from 'prop-types'
import { translate } from 'cozy-ui/transpiled/react/I18n'
import { Text, Caption } from 'cozy-ui/transpiled/react/Text'
import Icon from 'cozy-ui/transpiled/react/Icon'

import FieldsetTitle from '../Common/FieldsetTitle'
import Fieldset from '../Common/Fieldset'

const ACCOUNTS_MAPPING = {
  'konnector-google': {
    label: 'Google',
    icon: 'google'
  }
}

const ContactAccounts = ({ accounts, t }) => (
  <>
    <FieldsetTitle
      title={t('associated_accounts', { smart_count: accounts.length })}
    />
    <Fieldset>
      {accounts.map(account => {
        const { icon, label } = ACCOUNTS_MAPPING[account.type]
        return (
          <li key={account.id}>
            <div className="u-flex u-mt-1 u-mb-half">
              <div className="u-mr-1">
                <Icon icon={icon} />
              </div>
              <div>
                <div className="u-mb-half u-breakword">
                  <Text>{label}</Text>
                  <Caption>{account.name}</Caption>
                </div>
              </div>
            </div>
          </li>
        )
      })}
    </Fieldset>
  </>
)

ContactAccounts.propTypes = {
  accounts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired
    })
  )
}

export default translate()(ContactAccounts)
