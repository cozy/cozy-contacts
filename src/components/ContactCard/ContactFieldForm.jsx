import React from "react";
import PropTypes from "prop-types";
import { Field } from "react-final-form";
import Icon from "cozy-ui/react/Icon";

const getInputComponent = inputType =>
  inputType === "textarea" ? "textarea" : "input";

class ContactFieldInput extends React.Component {
  state = {
    renderLabel: false
  };

  showLabel = () => {
    this.setState({
      renderLabel: this.props.withLabel
    });
  };

  hideLabelIfEmpty = e => {
    this.setState({
      renderLabel: e.target.value && this.props.withLabel
    });
  };

  render() {
    const { name, type, withLabel } = this.props;
    const { renderLabel } = this.state;

    return (
      <div className="contact-form__input-wrapper">
        <Field
          name={name}
          type={type}
          onFocus={this.showLabel}
          onBlur={this.hideLabelIfEmpty}
          component={getInputComponent(type)}
          className="contact-form__input"
        />
        {withLabel &&
          renderLabel && (
            <Field
              name={`${name}Label`}
              type="text"
              component="input"
              className="contact-form__label-input"
            />
          )}
      </div>
    );
  }
}
ContactFieldInput.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  withLabel: PropTypes.bool
};
ContactFieldInput.defaultProps = {
  withLabel: false
};

const ContactFieldForm = ({ icon, name, type, label, inputWithLabel }) => (
  <div className="contact-form__field">
    <label className="contact-form__label">
      {icon && (
        <Icon icon={icon} color="coolGrey" className="contact-form__icon" />
      )}
      {label}
    </label>
    <ContactFieldInput name={name} type={type} withLabel={inputWithLabel} />
  </div>
);
ContactFieldForm.propTypes = {
  icon: PropTypes.any, // shall be a SVG prop type
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  inputWithLabel: PropTypes.bool
};
ContactFieldForm.defaultProps = {
  icon: null,
  inputWithLabel: false
};

export default ContactFieldForm;
