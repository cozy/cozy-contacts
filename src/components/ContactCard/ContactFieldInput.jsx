import React from 'react'
import PropTypes from 'prop-types'
import { Field } from 'react-final-form'
import classnames from 'classnames'

const getInputComponent = inputType =>
  inputType === 'textarea' ? 'textarea' : 'input'

class ContactFieldInput extends React.Component {
  state = {
    isRenderingLabel: false,
    hasFocus: false
  }

  onFocus = () => {
    this.setState({
      isRenderingLabel: this.props.withLabel,
      hasFocus: true
    })
  }

  onMainInputBlur = e => {
    this.setState(state => ({
      ...state,
      isRenderingLabel: e.target.value && this.props.withLabel,
      hasFocus: false
    }))
  }

  onSecondaryInputBlur = () => {
    this.setState(state => ({
      ...state,
      hasFocus: false
    }))
  }

  render() {
    const {
      name,
      type,
      placeholder,
      required,
      withLabel,
      labelPlaceholder
    } = this.props
    const { isRenderingLabel, hasFocus } = this.state

    return (
      <div
        className={classnames('contact-form__input-wrapper', {
          'contact-form__input-wrapper--focused': hasFocus
        })}
      >
        <Field
          name={name}
          type={type}
          placeholder={placeholder}
          required={required}
          onFocus={this.onFocus}
          onBlur={this.onMainInputBlur}
          component={getInputComponent(type)}
          className="contact-form__input"
        />
        {withLabel &&
          isRenderingLabel && (
            <Field
              name={`${name}Label`}
              type="text"
              component="input"
              className="contact-form__label-input"
              placeholder={labelPlaceholder}
              onFocus={this.onFocus}
              onBlur={this.onSecondaryInputBlur}
            />
          )}
      </div>
    )
  }
}
ContactFieldInput.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  withLabel: PropTypes.bool,
  labelPlaceholder: PropTypes.string,
  required: PropTypes.bool
}
ContactFieldInput.defaultProps = {
  withLabel: false,
  required: false,
  placeholder: '',
  labelPlaceholder: ''
}

export default ContactFieldInput
