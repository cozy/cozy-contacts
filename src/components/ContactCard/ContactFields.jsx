import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'cozy-ui/react/Icon'
import palette from 'cozy-ui/stylus/settings/palette.json'
import { translate } from 'cozy-ui/react/I18n'

import IconBirthday from '../../assets/icons/calendar.svg'
import IconNote from '../../assets/icons/comment.svg'
import IconCompany from '../../assets/icons/company.svg'
import IconCozy from '../../assets/icons/cozy.svg'
import IconEmail from '../../assets/icons/email.svg'
import IconFlag from '../../assets/icons/flag.svg'
import IconAddress from '../../assets/icons/location.svg'
import IconPhone from '../../assets/icons/phone-number.svg'

const ContactFields = ({ fields, title }) => (
  <div>
    {title && <h3 className="contact-fields-title">{title}</h3>}
    <ol className="contact-field-list">
      {fields.filter(field => field.values.length > 0).map((field, index) => (
        <li key={index}>
          <ContactField type={field.type} values={field.values} />
        </li>
      ))}
    </ol>
  </div>
)

ContactFields.propTypes = {
  fields: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.string.isRequired,
      values: PropTypes.array.isRequired
    })
  ),
  title: PropTypes.string
}

const ContactField = ({ type, values }) => (
  <div className="contact-field">
    <div className="contact-field-icon">
      <Icon icon={getIcon(type)} color={palette['coolGrey']} />
    </div>
    <div>
      {values.map((value, index) => (
        <FieldValueWithI18n type={type} value={value} key={index} />
      ))}
    </div>
  </div>
)

ContactField.propTypes = {
  type: PropTypes.string.isRequired,
  values: PropTypes.array.isRequired
}

const FieldValue = ({ type, value, t, f }) => {
  const renderedValue = renderFieldValue(value, type, t, f)
  const label = value.type || value.label || null

  return (
    <div className="contact-field-value">
      {renderedValue}
      {label && <span className="contact-field-separator">Â·</span>}
      {label && <span className="contact-field-label">{label}</span>}
    </div>
  )
}

FieldValue.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.object
  ]),
  t: PropTypes.func.isRequired,
  f: PropTypes.func.isRequired
}

const FieldValueWithI18n = translate()(FieldValue)

const iconsByType = {
  birthday: IconBirthday,
  note: IconNote,
  company: IconCompany,
  cozy: IconCozy,
  email: IconEmail,
  address: IconAddress,
  phone: IconPhone
}

const getIcon = fieldType => iconsByType[fieldType] || IconFlag

const emptyAddress = {
  street: '',
  pobox: '',
  city: '',
  region: '',
  postcode: '',
  country: ''
}

const renderFieldValue = (value, type, t, f) => {
  if (!value) return false
  if (type === 'birthday') return f(new Date(value), 'YYYY-M-D')
  if (typeof value !== 'object') return value.toString()

  switch (type) {
    case 'address':
      return value.formattedAddress
        ? value.formattedAddress
        : t('formatted_address', { ...emptyAddress, ...value }).trim()
    case 'email':
      return <a href={`mailto:${value.address}`}>{value.address}</a>
    case 'phone':
      return <a href={`tel:${value.number}`}>{value.number}</a>
    case 'cozy':
      return value.url
    default:
      return Object.keys(value)
        .map(label => `${label}: ${value[label]}`)
        .join(', ')
  }
}

export default ContactFields
