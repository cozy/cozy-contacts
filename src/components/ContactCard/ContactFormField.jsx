import React from "react";
import PropTypes from "prop-types";
import { FieldArray } from "react-final-form-arrays";
import Icon from "cozy-ui/react/Icon";
import palette from "cozy-ui/stylus/settings/palette.json";
import IconPlus from "../../assets/icons/small-plus.svg";
import IconCross from "../../assets/icons/small-cross.svg";

class ContactFormField extends React.Component {
  render() {
    const { name, icon, label, isArray, renderInput } = this.props;
    return (
      <div className="contact-form__field">
        <label className="contact-form__label">
          {icon && (
            <Icon
              icon={icon}
              color={palette["coolGrey"]}
              className="contact-form__icon"
            />
          )}
          {label}
        </label>
        {isArray ? (
          <FieldArray name={name}>
            {({ fields }) => (
              <div className="contact-form__inputs-wrapper">
                {fields.map((nameWithIndex, index) =>
                  this.renderArrayField(
                    fields,
                    index,
                    nameWithIndex,
                    name,
                    renderInput
                  )
                )}
              </div>
            )}
          </FieldArray>
        ) : (
          <div className="contact-form__inputs-wrapper">
            {renderInput(name)}
          </div>
        )}
      </div>
    );
  }

  addField = fields => fields.push(undefined);

  removeField = (fields, index) => {
    const isLastRemainignField = fields.length === 1;
    fields.remove(index);
    if (isLastRemainignField) this.addField(fields);
  };

  renderArrayField = (fields, index, nameWithIndex, name, renderInput) => {
    const isLastField = index === fields.length - 1;
    const hasValue = fields.value[index] && fields.value[index][name];
    const canAddField = isLastField && hasValue;

    return (
      <div className="contact-form__meta-wrapper" key={nameWithIndex}>
        {renderInput(`${nameWithIndex}.${name}`)}

        {hasValue && (
          <button
            type="button"
            onClick={() => this.removeField(fields, index)}
            className="contact-form__meta-button contact-form__meta-button--remove"
          >
            <Icon icon={IconCross} color={palette["coolGrey"]} />
          </button>
        )}

        {canAddField && (
          <button
            type="button"
            onClick={() => this.addField(fields)}
            className="contact-form__meta-button contact-form__meta-button--add"
          >
            <Icon icon={IconPlus} color="white" />
          </button>
        )}
      </div>
    );
  };
}
ContactFormField.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.any, // shall be a SVG prop type
  label: PropTypes.string.isRequired,
  isArray: PropTypes.bool,
  renderInput: PropTypes.func.isRequired
};
ContactFormField.defaultProps = {
  icon: null,
  isArray: false
};

export default ContactFormField;
