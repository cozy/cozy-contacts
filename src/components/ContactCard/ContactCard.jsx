import React from "react";
import PropTypes from "prop-types";
import ContactIdentity from "./ContactIdentity";
import ContactFields from "./ContactFields";
import { fullContactPropTypes } from "../ContactPropTypes";
import {
  getFieldListFrom,
  filterFieldList,
  groupUnsupportedFields,
  supportedFieldsInOrder,
  orderFieldList,
  makeValuesArray
} from "../../helpers/contacts";

const ContactCard = ({ title, contact, renderHeader, renderBody }) => {
  const fields = getFieldListFrom(contact);
  const filteredFields = filterFieldList(fields);
  const groupedFields = groupUnsupportedFields(
    filteredFields,
    supportedFieldsInOrder
  );
  const orderedFields = orderFieldList(groupedFields, supportedFieldsInOrder);
  const normalizedFields = makeValuesArray(orderedFields);

  return (
    <div>
      {renderHeader(<ContactIdentity contact={contact} />)}
      {renderBody(<ContactFields fields={normalizedFields} title={title} />)}
    </div>
  );
};

ContactCard.propTypes = {
  contact: fullContactPropTypes.isRequired,
  title: PropTypes.string.isRequired,
  renderHeader: PropTypes.func,
  renderBody: PropTypes.func
};

ContactCard.defaultProps = {
  renderHeader: children => children,
  renderBody: children => children
};

export default ContactCard;
