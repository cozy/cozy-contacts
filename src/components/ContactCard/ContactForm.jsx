import React from "react";
import PropTypes from "prop-types";
import { Form } from "react-final-form";
import arrayMutators from "final-form-arrays";
import ContactFormField from "./ContactFormField";
import ContactFieldInput from "./ContactFieldInput";
import { Button } from "cozy-ui/react/Button";
import { translate } from "cozy-ui/react/I18n";

import IconEmail from "../../assets/icons/email.svg";
import IconPhone from "../../assets/icons/phone-number.svg";
import IconAddress from "../../assets/icons/location.svg";
import IconCozy from "../../assets/icons/cozy.svg";
import IconCompany from "../../assets/icons/company.svg";
import IconBirthday from "../../assets/icons/calendar.svg";
import IconNote from "../../assets/icons/comment.svg";

const fields = [
  {
    name: "givenName",
    icon: null,
    type: "text"
  },
  {
    name: "familyName",
    icon: null,
    type: "text",
    required: true
  },
  {
    name: "phone",
    icon: IconPhone,
    type: "tel",
    hasLabel: true,
    isArray: true
  },
  {
    name: "email",
    icon: IconEmail,
    type: "email",
    hasLabel: true,
    isArray: true
  },
  {
    name: "address",
    icon: IconAddress,
    type: "text",
    hasLabel: true,
    isArray: true
  },
  {
    name: "cozy",
    icon: IconCozy,
    type: "url",
    hasLabel: true
  },
  {
    name: "company",
    icon: IconCompany,
    type: "text"
  },
  {
    name: "birthday",
    icon: IconBirthday,
    type: "date"
  },
  {
    name: "note",
    icon: IconNote,
    type: "textarea"
  }
];

// initialize the form values, required so that array fields start with at least one editable field
const initialFieldValues = fields.reduce((initialValues, { name, isArray }) => {
  initialValues[name] = isArray ? [undefined] : undefined;
  return initialValues;
}, {});

class ContactForm extends React.Component {
  formDataToContact = data => {
    const {
      givenName,
      familyName,
      phone,
      email,
      address,
      cozy,
      company,
      birthday,
      note
    } = data;

    const fullName = (givenName || "") + " " + (familyName || "");

    const contact = {
      fullname: fullName.trim(),
      name: {
        givenName,
        familyName
      },
      email: email.filter(val => val).map(({ email, emailLabel }, index) => ({
        address: email,
        type: emailLabel,
        primary: index === 0
      })),
      address: address
        .filter(val => val)
        .map(({ address, addressLabel }, index) => ({
          formattedAddress: address,
          type: addressLabel,
          primary: index === 0
        })),
      phone: phone.filter(val => val).map(({ phone, phoneLabel }, index) => ({
        number: phone,
        type: phoneLabel,
        primary: index === 0
      })),
      groups: [],
      cozy: cozy
        ? [
            {
              url: cozy,
              label: data["cozyLabel"],
              primary: true
            }
          ]
        : undefined,
      company,
      birthday,
      note,
      metadata: {
        version: 1
      }
    };

    this.props.onSubmit(contact);
  };

  render() {
    const { onCancel, t } = this.props;
    return (
      <Form
        mutators={{ ...arrayMutators }}
        onSubmit={this.formDataToContact}
        initialValues={initialFieldValues}
        render={({ handleSubmit }) => (
          <div>
            <form onSubmit={handleSubmit} className="contact-form">
              <div className="contact-form__fields">
                {fields.map(
                  ({ name, icon, type, required, hasLabel, isArray }) => (
                    <ContactFormField
                      key={name}
                      name={name}
                      icon={icon}
                      label={t(`field.${name}`)}
                      isArray={isArray}
                      renderInput={inputName => (
                        <ContactFieldInput
                          name={inputName}
                          type={type}
                          placeholder={t(`placeholder.${name}`)}
                          required={required}
                          withLabel={hasLabel}
                          labelPlaceholder={t("placeholder.label")}
                        />
                      )}
                    />
                  )
                )}
              </div>

              <div className="contact-form__footer">
                <div>
                  <Button
                    type="button"
                    theme="secondary"
                    label={t("cancel")}
                    onClick={onCancel}
                  />
                  <Button type="submit" label={t("save")} />
                </div>
              </div>
            </form>
          </div>
        )}
      />
    );
  }
}

ContactForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};

export default translate()(ContactForm);
