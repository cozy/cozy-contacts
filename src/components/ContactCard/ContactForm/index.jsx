import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'react-final-form'
import arrayMutators from 'final-form-arrays'
import { translate } from 'cozy-ui/transpiled/react/I18n'

import FieldsetTitle from '../../Common/FieldsetTitle'
import ContactFormField from './ContactFormField'
import ContactFieldInput from '../ContactFieldInput'
import { fullContactPropTypes } from '../../ContactPropTypes'
import contactToFormValues from './contactToFormValues'
import formValuesToContact from './formValuesToContact'
import { fields } from './fieldsConfig'

// this variable will be set in the form's render prop
// and used by the submit button in ContactFormModal
// to be able to trigger the submit from outside the form
// See react-final-form examples here: https://www.npmjs.com/package/react-final-form#external-submit
let _submitContactForm

function setSubmitContactForm(handleSubmit) {
  _submitContactForm = handleSubmit
}

export function getSubmitContactForm() {
  return _submitContactForm
}

const ContactForm = ({ contact, onSubmit, t }) => (
  <Form
    mutators={{ ...arrayMutators }}
    onSubmit={data => onSubmit(formValuesToContact(data, contact))}
    initialValues={contactToFormValues(contact, t)}
    render={({ handleSubmit }) => {
      setSubmitContactForm(handleSubmit)
      return (
        <div>
          <form onSubmit={handleSubmit} className="u-flex u-flex-column">
            <FieldsetTitle title={t('contact_info')} />
            {fields.map(
              ({
                name,
                icon,
                type,
                required,
                hasLabel,
                isArray,
                labelProps,
                isMultiline
              }) => (
                <ContactFormField
                  key={name}
                  name={name}
                  icon={icon}
                  label={t(`fields.${name}`)}
                  t={t}
                  isArray={isArray}
                  renderInput={(inputName, id) => (
                    <ContactFieldInput
                      id={id}
                      name={inputName}
                      type={type}
                      label={t(`fields.${name}`)}
                      required={required}
                      withLabel={hasLabel}
                      labelPlaceholder={t('fields.label')}
                      labelProps={labelProps}
                      isMultiline={isMultiline}
                    />
                  )}
                />
              )
            )}
          </form>
        </div>
      )
    }}
  />
)

ContactForm.propTypes = {
  contact: fullContactPropTypes,
  onSubmit: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
}

export default translate()(ContactForm)
