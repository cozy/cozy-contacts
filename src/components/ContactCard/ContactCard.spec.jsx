import React from "react";
import renderer from "react-test-renderer";
import ContactCard from "./ContactCard";
import { I18n } from "cozy-ui/react/I18n";

describe("ContactCard", () => {
  test("should match snapshot", () => {
    const contact = {
      _id: "c6899688-6cc6-4ffb-82d4-ab9f9b82c582",
      _rev: "1-9368a4f2e467c449f4a1f5171a784aa8",
      address: [
        {
          city: "Stafford",
          country: "Solomon Islands",
          postcode: "06635",
          primary: true,
          street: "48 Fuller Road"
        }
      ],
      birthday: "1998-06-03",
      email: [
        { address: "rikki.white@celmax.name", primary: true },
        { address: "eleanore.fennell@thermolock.name", primary: false }
      ],
      name: { familyName: "White", givenName: "Rikki" },
      phone: [
        { number: "+33 (1)9 14 02 28 31", primary: true },
        { number: "+33 (2)3 99 53 65 21", primary: false }
      ],
      unsupportedField: "unsupportedField"
    };
    const tree = renderer
      .create(
        <I18n lang="en" dictRequire={() => ""}>
          <ContactCard contact={contact} title="title" />
        </I18n>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
