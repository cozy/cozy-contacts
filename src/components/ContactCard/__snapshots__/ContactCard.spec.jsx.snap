// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should match snapshot 1`] = `
<div>
  <div
    className="contact-card-identity"
  >
    <div
      className="c-avatar c-avatar--medium"
      style={
        Object {
          "backgroundColor": "#FD7461",
          "color": "white",
        }
      }
    >
      
      <span
        className="c-avatar-initials"
      >
        RW
      </span>
    </div>
    <div
      className="contact-card-identity__infos"
    >
      <h1
        className="contact-card-identity__title"
      >
        Rikki  White
      </h1>
      <div
        className="contact-card-identity__groups"
      >
        <div
          className="react-select__container css-1c6nt7d"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="true"
            aria-live="polite"
            className="css-17l7ekz"
            role="status"
          >
            0 results available.
          </span>
          <div
            innerRef={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <button
              aria-busy={false}
              className="c-btn c-btn--secondary c-btn--small"
              disabled={false}
              onClick={null}
              role="button"
              type="submit"
            >
              <span>
                <svg
                  className="icon"
                  height="16"
                  style={
                    Object {
                      "fill": "#95999D",
                    }
                  }
                  width="16"
                >
                  <use
                    xlinkHref="#icon-0"
                  />
                </svg>
                groups.manage
                <svg
                  className="icon group-manager__indicator"
                  height="16"
                  style={
                    Object {
                      "fill": "#95999D",
                    }
                  }
                  width="12"
                >
                  <use
                    xlinkHref="#icon-0"
                  />
                </svg>
              </span>
            </button>
            <div
              className="select-control__input"
            >
              <div
                className="react-select__value-container react-select__value-container--isMulti css-6rkci4"
              >
                <div
                  className="react-select__placeholder css-16ucli7"
                >
                  Select...
                </div>
                <input
                  className="css-hg63qv"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  readOnly={true}
                  value=""
                />
              </div>
              <div
                className="react-select__indicators css-1lfa1d8"
              >
                <span
                  className="react-select__indicator-separator css-1cnd6ox"
                  css={undefined}
                  role="presentation"
                />
                <div
                  className="react-select__indicator react-select__dropdown-indicator css-1c0gw5t"
                  css={undefined}
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  role="button"
                >
                  <svg
                    className="react-select__icon react-select__down-icon css-qyb9b"
                    focusable="false"
                    height={20}
                    role="presentation"
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ol
          className="contact-groups-list"
        />
      </div>
    </div>
  </div>
  <div>
    <h3
      className="contact-fields-title"
    >
      title
    </h3>
    <ol
      className="contact-field-list"
    >
      <li>
        <div
          className="contact-field"
        >
          <div
            className="contact-field-icon"
          >
            <svg
              className="icon"
              height="16"
              style={
                Object {
                  "fill": "#95999D",
                }
              }
              width="16"
            >
              <use
                xlinkHref="#icon-0"
              />
            </svg>
          </div>
          <div>
            <div
              className="contact-field-value"
            >
              <a
                href="tel:+33 (1)9 14 02 28 31"
              >
                +33 (1)9 14 02 28 31
              </a>
            </div>
            <div
              className="contact-field-value"
            >
              <a
                href="tel:+33 (2)3 99 53 65 21"
              >
                +33 (2)3 99 53 65 21
              </a>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="contact-field"
        >
          <div
            className="contact-field-icon"
          >
            <svg
              className="icon"
              height="16"
              style={
                Object {
                  "fill": "#95999D",
                }
              }
              width="16"
            >
              <use
                xlinkHref="#icon-0"
              />
            </svg>
          </div>
          <div>
            <div
              className="contact-field-value"
            >
              <a
                href="mailto:rikki.white@celmax.name"
              >
                rikki.white@celmax.name
              </a>
            </div>
            <div
              className="contact-field-value"
            >
              <a
                href="mailto:eleanore.fennell@thermolock.name"
              >
                eleanore.fennell@thermolock.name
              </a>
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="contact-field"
        >
          <div
            className="contact-field-icon"
          >
            <svg
              className="icon"
              height="16"
              style={
                Object {
                  "fill": "#95999D",
                }
              }
              width="16"
            >
              <use
                xlinkHref="#icon-0"
              />
            </svg>
          </div>
          <div>
            <div
              className="contact-field-value"
            >
              formatted_address
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="contact-field"
        >
          <div
            className="contact-field-icon"
          >
            <svg
              className="icon"
              height="16"
              style={
                Object {
                  "fill": "#95999D",
                }
              }
              width="16"
            >
              <use
                xlinkHref="#icon-0"
              />
            </svg>
          </div>
          <div>
            <div
              className="contact-field-value"
            >
              1998-6-3
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="contact-field"
        >
          <div
            className="contact-field-icon"
          >
            <svg
              className="icon"
              height="16"
              style={
                Object {
                  "fill": "#95999D",
                }
              }
              width="16"
            >
              <use
                xlinkHref="#icon-0"
              />
            </svg>
          </div>
          <div>
            <div
              className="contact-field-value"
            >
              unsupportedField
            </div>
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
